util/httputil/http_util.go

```diff
@@ -67,7 +65,19 @@ func newAuthRoundTripper(auth *config.Auth, rt http.RoundTripper) http.RoundTrip
 	}
 }
 
-func AuthTransport(auth *config.Auth, transportConfig *http.Transport) (http.RoundTripper, error) {
+func CreateTransport(auth *config.Auth, transportConfig *http.Transport, timeout time.Duration) (http.RoundTripper, error) {
+	// Limits the time spent establishing a TCP connection if a new one is
+	// needed.
+	transportConfig.Dial = (&net.Dialer{
+		Timeout:   timeout,
+		KeepAlive: timeout,
```

Comment:
Default value for KeepAlive is 15 seconds.
TBH, I think setting `KeepAlive` and `Timeout` doesn't make much sense. Connection will anyway be terminated while the KeepAlive probe is on it's way. Probably, KeepAlive is more useful in long lasting connections.

So, I think we can omit KeepAlive from this PR and let the default of 15 seconds to take effect if the transport is created with longer timeouts. The KeepAlive probes will detect the broken connection and, hopefully, that will trigger an error.

Reply:
> TBH, I think setting `KeepAlive` and `Timeout` doesn't make much sense.

More context: setting both to the same value :)